server/SocketManager/socketmanager.js

```diff
@@ -1,11 +1,14 @@
 let sockets = {};
+let socket_user_map = {};
 
-const updateSocket = (usertoken, clientsocket) => {
+const updateSocket = (usertoken, clientsocket, user_id) => {
   sockets[usertoken] = clientsocket;
+  socket_user_map[usertoken] = user_id;
 };
 
 const removeSocket = usertoken => {
   delete sockets[usertoken];
+  delete socket_user_map[usertoken];
```

Comment:
In the `removeSocket` function, you are deleting entries from both `sockets` and `socket_user_map`. Ensure that the logic for removing sockets is consistent and that both maps are always in sync. It might be beneficial to add error handling or logging to track if a user token does not exist in either map.


    *This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*